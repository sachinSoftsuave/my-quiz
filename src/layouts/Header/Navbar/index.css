.linkHighScore{
     cursor:pointer;
     text-decoration: none;
     font-weight:900px;
     font-family: Verdana, Geneva, Tahoma, sans-serif;
     font-size: 15px;
     color:white;
 }