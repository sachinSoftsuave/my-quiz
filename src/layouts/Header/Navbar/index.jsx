import React from "react";
import './index.css'

function Navbar() {
  return (
    <div>
      <a className="linkHighScore" href="./highScore">
        View Highscore <i className="fas fa-hand-point-left fa-lg"></i>
      </a>
    </div>
  );
}

export default Navbar;
