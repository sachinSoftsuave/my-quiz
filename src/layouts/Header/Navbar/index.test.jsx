import React from "react";
import { mount } from "enzyme";
import toJson from 'enzyme-to-json';

import Navbar from "./index.jsx";


describe("renders navbar component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<Navbar />);
  });

  it("renders the text of view score", () => {
    expect(wrapper.find("a").text()).toContain("View Highscore ");
  });

  it('when view highscore is clicked it navigates to highscore page', () => {
      wrapper.find('a').simulate('click')
    wrapper.setProps({ articles: [{Url : './high_Score'}]})
    expect(wrapper.find('a').prop('href')).toEqual('./high_Score')
  })

  it("should test Navbar component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should have icon", () => {
    expect(wrapper.find("i")).toHaveLength(1);
  });
});
