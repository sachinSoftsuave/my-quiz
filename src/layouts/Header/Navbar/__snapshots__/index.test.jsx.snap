// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders navbar component should test Navbar component 1`] = `
<Navbar>
  <div>
    <a
      className="linkHighScore"
      href="./highScore"
    >
      View Highscore 
      <i
        className="fas fa-hand-point-left fa-lg"
      />
    </a>
  </div>
</Navbar>
`;
